virtualbox.py

```diff
@@ -0,0 +1,78 @@
+import subprocess
+from pathlib import Path
+from sys import platform
+import datetime
+from pytz import UTC
+
+utc = UTC
+
+class virtualbox(object):
+
+    def __init__(self, virtualboxpath, tz) -> None:
+        if platform.lower() == "win32":
+            virtualboxpath = virtualboxpath.replace('\"', "")
+            virtualboxpath = virtualboxpath.replace("\'", "")
+            self.vmrunpath = Path(virtualboxpath).joinpath("VBoxManage.exe")
+        else:
+            self.vmrunpath = virtualboxpath
+        self.tz = tz
+        self.vminfo = []
+        self.output = []
+
+    def updateOutput(self) -> None:
+        # output = subprocess.run([str(self.vmrunpath), "list", "runningvms"], stdout=subprocess.PIPE)
+        # output = output.stdout.decode("utf-8")
+        with subprocess.Popen([str(self.vmrunpath), "list", "runningvms"], stdout=subprocess.PIPE) as proc:
```

Comment:
I don't think Popen has a noticeable benefit over run here, but it doesn't really affect anything so. There shouldn't be any zombie processes, VBoxManage just prints data and exits

Reply:
I am probably going to investigate this sometime. For now i reverted back to the run function, although i think the use of context managers is overall preferable